### ISSUE TITLE
`--unwindlib=none` warning during Rust build script compilation

### ISSUE THREAD

**Issue by robbie-vanderzee** [cerisier/toolchains_llvm_bootstrapped#304]
Our build is working great with `0.5.7` (thank you for the help on #271!), but we're seeing this warning on every C compilation triggered by Rust build scripts (e.g. `tree-sitter`):

```
clang++: warning: argument unused during compilation: '--unwindlib=none' [-Wunused-command-line-argument]
```

The toolchain correctly scopes `--unwindlib=none` to link actions in `toolchain/args/BUILD.bazel`, but Rust build scripts using the `cc` crate receive it via `LDFLAGS` and pass it to both compile and link invocations.

Is there a way to suppress this, or is it expected?

**Comment by basvandijk** [comment]
We're [switching](https://github.com/dfinity/ic/pull/10991) (from hermetic_cc_toolchain) to hermetic-llvm and noticed the above mentioned warnings in our logs when building rust crates (using rules_rust):
```
[1,922 / 2,024] Compiling Rust rlib prettyplease v0.2.37 (29 files) [for tool]; 7s remote-cache, linux-sandbox ... (12 actions, 11 running)
[1,942 / 2,024] Compiling Rust proc-macro async_trait v0.1.89 (27 files) [for tool]; 9s remote-cache, linux-sandbox ... (17 actions, 16 running)
[1,962 / 2,024] Compiling Rust proc-macro async_trait v0.1.89 (27 files) [for tool]; 10s remote-cache, linux-sandbox ... (13 actions, 12 running)
INFO: From Compiling Rust proc-macro ic-heap-bytes-derive v0.1.0 (1 file) [for tool]:
warning: linker stderr: clang++: argument unused during compilation: '--unwindlib=none' [-Wunused-command-line-argument]
  |
  = note: requested on the command line with `-W linker-messages`

warning: 1 warning emitted
```
We're working around this by patching hermetic-llvm with: [`bazel/hermetic_llvm_unwindlib.patch`](https://github.com/dfinity/ic/blob/12f84ae40ccf39eb785632618a9080c8654d01a6/bazel/hermetic_llvm_unwindlib.patch).

**Comment by cerisier** [comment]
Thanks ! rules_rust right ? not rules_rs ?

**Comment by cerisier** [comment]
Ok so as explained in https://github.com/hermeticbuild/rules_rust/pull/38,

The only beautiful way to solve this is to find a solution for https://github.com/bazelbuild/rules_cc/issues/722.
Then we can implement https://github.com/hermeticbuild/hermetic-llvm/pull/564 and this issue will disappear.

Meanwhile, maybe we should provide a flag specific for that problem that convey the fact that a conflicting hermetic flag is added later by a third party command (like rustc for --nodefaultlibs). While I'm writing this, I also realize that --unwindlib=none LDFLAGS only make sense if cc_common.link is used, so maybe a flag that says, `cc_common_link not used downstream` ? 

WDYT @dzbarsky ?
